### PR TITLE
[docs] replaced sphinx-bootstrap-theme with sphinx-rtd-theme

### LINKED ISSUE #158: [Doc] when jumping to anchors, the first lines are covered by the top bar (Firefox)
When jumping to anchors inside the documentation like (https://kafe2.readthedocs.io/en/latest/parts/developer_guide.html#tools), the heading and the first line are covered up by the top bar.

This does happen on Firefox 80 (Gecko), but not on Gnome Web 3.36.4 (WebKit).

So this might or might not be a browser related issue, and testing with more browsers should be done (chromium?).
Can we fox it on our end with some css improvements? I am no web developer and have no clue how to deal with this.

### PR DESCRIPTION
**Pull request by MitchiLaser**
This pull request contains many commits which replace the sphinx-bootstrap-theme with the sphinx-rtd-theme. Furthermore a lot of files which were located in the configuration directory for Sphinx were removed because they were never needed.

Please squash all the commits when accepting the pull request, they are still separated in case a wrong file was deleted.

### CODE REVIEW

**Comment by MitchiLaser** [comment]
Fixes #158

File: doc/src/conf.py [hunk 1]
```diff
@@ -51,17 +49,25 @@
...
+import tomllib
+with open("../../pyproject.toml", "rb") as f:
+    toml = tomllib.load(f)
 
-# style sheet customizations
-def setup(app):
-    app.add_css_file("style.css")
+pyproject = toml["project"]
+# extract list of authors from pyproject.toml and replace the first name by its first letter
+authors = ", ".join([i['name'].split(' ')[0][0] + ". " + i['name'].split(' ')[1] for i in pyproject['authors']])
```
Comment (cverstege) [inline doc/src/conf.py]: You might want to use the last part as last name. E.g. if there are multiple first names
```suggestion
authors = ", ".join([i['name'].split(' ')[0][0] + ". " + i['name'].split(' ')[-1] for i in pyproject['authors']])
```

File: pyproject.toml [hunk 2]
```diff
@@ -6,11 +6,11 @@ build-backend = "setuptools.build_meta"
...
 authors = [
-	{ name="Daniel Savoiu", email="daniel.savoiu@cern.ch" },
 	{ name="Johannes Gäßler", email="johannes.gaessler@cern.ch" },
-	{ name="Cedric Verstege", email="cedric.verstege@cern.ch" },
-	{ name="Günter Quast", email="G.Quast@kit.edu" },
 	{ name="Michael Hohenstein", email="michael@hohenste.in" },
+	{ name="Günter Quast", email="G.Quast@kit.edu" },
+	{ name="Daniel Savoiu", email="daniel.savoiu@cern.ch" },
+	{ name="Cedric Verstege", email="cedric.verstege@cern.ch" },
 ]
```
Comment (cverstege) [inline pyproject.toml:15]: Maybe add a comment about alphabetical ordering.

**Review by cverstege — CHANGES_REQUESTED** [review]
Looks good. Only two minor comments inline.

File: pyproject.toml (hunk 2 above)
Comment (MitchiLaser) [reply pyproject.toml:15]: I added a comment above the list of authors

File: doc/src/conf.py (hunk 1 above)
Comment (MitchiLaser) [reply doc/src/conf.py]: Done. This line might still not work perfectly for every case but it is getting the job done for the current list of maintainers.

File: doc/src/conf.py [hunk 3]
```diff
@@ -92,23 +97,23 @@ def setup(app):
                        'iminuit': ('https://iminuit.readthedocs.io/en/latest', None)}
 
 # Add any paths that contain templates here, relative to this directory.
-templates_path = ['_templates']
+# the '_templates' directory was existing in the past but the template in there wasn't used at all.
```
Comment (JohannesGaessler) [inline doc/src/conf.py]:
```suggestion
# the '_templates' directory exited in the past but the template in there wasn't used at all.
```

File: pyproject.toml [hunk 4]
```diff
@@ -5,12 +5,13 @@ build-backend = "setuptools.build_meta"
 [project]
 name = "kafe2"
 description = "Karlsruhe Fit Environment 2: a package for fitting and elementary data analysis"
+# the list of authors is ordered alphabetically by the last name
```
Comment (JohannesGaessler) [inline pyproject.toml]:
```suggestion
# the list of authors is ordered alphabetically by last name
```

**Review by JohannesGaessler — APPROVED** [review]

**Pull request merged**